### ISSUE TITLE
CalEITC investment income limit not applied correctly

### ISSUE THREAD

**Issue by martinholmer** [PolicyEngine/policyengine-us#1905]
The `ca_eitc` variable logic is not working as intended in the 0.237.0 and 0.237.1 versions.  

The following integration test illustrates a problem that causes some of the largest differences between PolicyEngineUS and TAXSIM35.   The problem is that current logic allows a tax unit with substantial investment income to receive the CalEITC.

First I show the integration test case and then below that I show the test result showing more than a $533 error.

```
- name: Tax unit with taxsimid 95611 in e21.its.csv and e21.ots.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        is_tax_unit_head: true
        age: 68
        employment_income: 15010
        taxable_interest_income: 11010
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person2:
        age: 11
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person3:
        age: 11
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
    spm_units:
      spm_unit:
        members: [person1, person2, person3]
        snap: 0  # not in TAXSIM35
        tanf: 0  # not in TAXSIM35
    tax_units:
      tax_unit:
        members: [person1, person2, person3]
        premium_tax_credit: 0  # not in TAXSIM35
        ca_use_tax: 0  # not in TAXSIM35
    households:
      household:
        members: [person1, person2, person3]
        state_code: CA
  output:  # expected results from patched TAXSIM35 2023-02-22 version
    # expected US results:
    earned_income_tax_credit: 0  # too much investment income to be eligible
    income_tax: -9648
    # expected CA results:
    ca_income_tax_before_credits: 164.14  # this is a PolicyEngineUS output
    ca_exemptions: 1_058  # treated as a nonrefundable credit
    ca_income_tax_before_refundable_credits: 0
    ca_eitc: 0  # too much investment income to be CalEITC eligible
    ca_income_tax: 0
```

Here is the test result for this case:

```
================================= FAILURES ===================================
_______________________________ test session _________________________________
/Users/mrh/work/Policy-Engine-US/CA/e21-case.yaml:
  Test 'Tax unit with taxsimid 95611 in e21.its.csv and e21.ots.csv':
    ca_eitc@2021: [533.309] differs from 0.0 
                   with an absolute margin [533.309] > 0.01
========================= short test summary info ============================
FAILED e21-case.yaml::
============================ 1 failed in 1.26s ===============================

```

**Comment by MaxGhenis** [comment]
Thanks Martin. I just looked at the code and didn't see anything obviously amiss. @nikhilwoodruff could you take a look?

**Comment by nikhilwoodruff** [comment]
Yeah- just took a look, it's because the second phase-out region logic didn't take into account the eligibility criteria. Fixed in #1908
